ocaml/lambda/matching.ml

```diff
@@ -3887,58 +3925,75 @@ let compile_flattened ~scopes value_kind repr partial ctx pmh =
         (compile_match ~scopes value_kind repr partial)
         lam total ctx hs
 
-let do_for_multiple_match ~scopes value_kind loc paraml mode pat_act_list partial =
+let do_for_multiple_match ~scopes ~return_layout loc paraml mode pat_act_list partial =
+  (* CR layouts v5: This function is called in cases where the scrutinee of a
+     match is a literal tuple (e.g., [match e1, e2, e3 with ...]).  The
+     typechecker treats the scrutinee here like any other tuple, so it's fine to
+     assume the whole thing and the elements have sort value.  That will change
+     when we allow non-values in structures. *)
   let repr = None in
+  let param_lambda = List.map (fun (l, _, _) -> l) paraml in
   let arg =
     let sloc = Scoped_location.of_location ~scopes loc in
-    (* CR ncourant: this can build a mixed block, but it should never actually be
```

Comment:
My read of the parsing and type checking code is that this can't be a mixed block because the scrutinee of a "multiple match" (`match a, b, c with ...`) is just a normal tuple and typechecked as such.  I tried to reflect that in the new comment above.  But maybe I misread the type checker?

Reply:
Ah, good point, I misread your original comment.  It sounds like ncourant was uncertain about the block not being constructed, and I agree it might be middle-end dependent, so my instinct is not to risk leaving an inaccurate comment now that the CR content is resolved.